### PR TITLE
[android] Handle null pointers in Java

### PR DESCRIPTION
**Pull request by AJenbo**
Mostly minor issues. But the issue with accessing internal files have about 5 crashes related to it. Though I'm curious how you can end up with a device where the app doesn't have it's internal files folder... but hey we do not rely on it anyway.

### CODE REVIEW

File: android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java [hunk 1]
```diff
@@ -60,7 +63,7 @@ private void importFile(Uri fileUri) {
 		if (fileUri == null)
 			return;
 
-		DocumentFile file = DocumentFile.fromSingleUri(getApplicationContext(), fileUri);
+		DocumentFile file = Objects.requireNonNull(DocumentFile.fromSingleUri(getApplicationContext(), fileUri));
```
Comment (AJenbo) [inline android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:66]: fromSingleUri only returns null if called on versions older then KitKat which we already guard against accessing this function, so simply throwing a null pointer exception at this point should never cause an issue.

**Comment by tsunamistate** [comment]
@AJenbo is there a specific test case I should test for, or just do a smoke test? (Wow, that's a lot of "test")

**Comment by AJenbo** [comment]
Strictly code review is just to see if you think the code looks sensible. But I won't stop you from also testing things ;) Especially as there can be a lot of variety in Android devices.

The changes here affects:
- Using the hold-app-icon and selecting import files to import a save game or mpq file
- Launching the game with save games from the legacy app on the device
- Launching the app with out any mpq files

If any of that seams easy to test go ahead, but don't do anything you think is a hazel.

**Comment by tsunamistate** [comment]
Code review is NOT my thing, unfortunately. I am QA, not a programmer
Those cases are easy enough, I'll get back to you with the results soon

File: android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java [hunk 2]
```diff
@@ -20,6 +22,7 @@ public class ImportActivity extends Activity {
 
 	private static final int IMPORT_REQUEST_CODE = 0xD1AB70;
 
+	@RequiresApi(api = Build.VERSION_CODES.KITKAT)
```
Comment (yuripourre) [inline android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:25]: I didn't look into the entire code but having this annotation on the onCreate code will prevent the app to run on devices prior to this. Maybe the part that handles the file could be extracted to a new method and this annotation moved there.

Comment (AJenbo) [reply android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:25]: This is only accessible tought the shortcut list which is a feature that only exists in Android 25+, so preventing Android 18 from opening it isn't a problem.

File: android-project/app/build.gradle [hunk 3]
```diff
@@ -65,4 +65,5 @@ android {
 dependencies {
 	implementation fileTree(include: ['*.jar'], dir: 'libs')
 	implementation 'com.android.support.constraint:constraint-layout:2.0.4'
+	implementation 'org.jetbrains:annotations:15.0'
```
Comment (ephphatha) [inline android-project/app/build.gradle:68]: Should this be a compileOnly dependency? Doesn't look like it's needed at runtime.

Comment (AJenbo) [reply android-project/app/build.gradle:68]: Don't know much about Java, it was added automatically by Android Studio (Jetbrain) when i added the annotation suggestion. If I had know that it was the automatic fixes that you guys thought where controversial I would have just left them out :joy:

Comment (ephphatha) [reply android-project/app/build.gradle:68]: 😂 surprised jetbrain doesn't use their own recommended config

https://github.com/JetBrains/java-annotations

```suggestion
	compileOnly 'org.jetbrains:annotations:23.0.0'
```

Comment (AJenbo) [reply android-project/app/build.gradle:68]: It did also initially add them in there long form and without importing the namespace, had to google a bit to make it compile :roll_eyes:

File: android-project/app/build.gradle [hunk 4]
```diff
@@ -65,4 +65,5 @@ android {
 dependencies {
 	implementation fileTree(include: ['*.jar'], dir: 'libs')
 	implementation 'com.android.support.constraint:constraint-layout:2.0.4'
+	compileOnly 'org.jetbrains:annotations:23.0.0'
```
Comment (ephphatha) [inline android-project/app/build.gradle:68]: Looking at the imports in DataActivity/ImportActivity this package isn't being used, instead you're referencing android.support.annotation. The current version of that package is [androidx.annotation](https://mvnrepository.com/artifact/androidx.annotation/annotatio, not sure if it's a straightforward migration or something needs to be done to enable compiling for older targets?

Comment (AJenbo) [reply android-project/app/build.gradle]: Looks to work with out it, maybe I should just start ignoring there suggestions.

File: android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java [hunk 5]
```diff
@@ -87,7 +90,10 @@ private void importFile(Uri fileUri) {
 					outputStream.close();
 			}
 		} catch (IOException exception) {
-			Log.e("importFile", Objects.requireNonNull(exception.getMessage()));
+			String message = exception.getMessage();
+			if (message == null)
+				return;
+			Log.e("importFile", message);
```
Comment (yuripourre) [inline android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:96]: When message is null, this call shouldn´t break but maybe would be more informative to print the entire exception so will output the stacktrace.

```suggestion
if (message == null) {
	Log.e("importFile", exception);
} else {
	Log.e("importFile", message);
}
```
Or simpler, we can ignore the message and print the exception only.
```java
	Log.e("importFile", exception);
```

File: android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java (hunk 2 above)
Comment (yuripourre) [reply android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:25]: @AJenbo in that case, that's ok.

File: android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java (hunk 5 above)
Comment (AJenbo) [reply android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:96]: Thanks, this is my first time writing Java code :D

Comment (AJenbo) [reply android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java]: Apparently that isn't possible:

```
/home/runner/work/devilutionX/devilutionX/android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java:97: error: incompatible types: IOException cannot be converted to String

	Log.e("importFile", exception);
> Task :app:compileDebugJavaWithJavac FAILED
```

Comment (ephphatha) [reply android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java]: try `log.e("importFile", "IOException", exception);`? https://developer.android.com/reference/android/util/Log#e(java.lang.String,%20java.lang.String,%20java.lang.Throwable)

Comment (yuripourre) [reply android-project/app/src/main/java/org/diasurgical/devilutionx/ImportActivity.java]: @AJenbo sorry, @ephphatha is right, Log.e requires a third param to print an exception.
```java
if (message == null) {
	Log.e("importFile", "IOException", exception);
} else {
	Log.e("importFile", message);
}
```

File: android-project/app/src/main/java/org/diasurgical/devilutionx/DataActivity.java [hunk 6]
```diff
@@ -9,6 +9,7 @@
 import android.database.Cursor;
 import android.net.Uri;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
```
Comment (yuripourre) [inline android-project/app/src/main/java/org/diasurgical/devilutionx/DataActivity.java:12]: This import is better than the one suggested by IntelliJ, indeed.

**Review by yuripourre — APPROVED** [review]

**Pull request merged**